vignettes/adrs_gcig.Rmd

```diff
@@ -0,0 +1,811 @@
+  pull(warn)
+
+if (length(covr_r11_check) > 0) {
+  cli::cli_warn("{covr_r11_check}")
+}
+
+covr_r11 <- covr_r11 %>%
+  select(-AVAL, -AVALC, -AVAL.next, -AVALC.next, -ADT.next) %>%
+  rename(AVALC = AVALC.confirmed) %>%
+  mutate(AVAL = aval_resp(AVALC)) %>%
+  mutate(
+    PARAMCD = "COVRR11",
+    PARAM = "Confirmed RECIST 1.1 Overall Response by Investigator",
+    PARAMN = 7,
+    PARCAT1 = "RECIST 1.1",
+    PARCAT1N = 2,
+    PARCAT2 = "Investigator",
+    PARCAT2N = 1,
+  )
+
+adrs <- bind_rows(adrs, covr_r11)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT),
+  filter = PARAMCD == "COVRR11" & AVALC != "MISSING"
+)
+```
+
+## Combined Best Confirmed Overall Response {#combcbor}
+
+Now that we have both components of the combined response as confirmed responses, we can derive the combined response and at a later step Combined Best Confirmed Overall Response parameter.
+
+Following table provides a summary of the `Combined Overall Response by Investigator (Confirmation of RECIST1.1 Reqired)` calculation at each time point as per GCIG guidelines.
+
+```{r echo=FALSE}
+list_resp <- tribble(
+  ~"Confirmed RECIST1.1 response", ~"Confirmed CA-125 response", ~"Combined response",
+  "CR", "CR", "CR",
+  "CR", "PR/SD/NE", "PR",
+  "PR/SD/NE", "CR", "PR",
+  "PR", "CR/PR/SD/NE", "PR",
+  "CR/PR/SD/NE", "PR", "PR",
+  "SD", "SD", "SD",
+  "SD", "NE", "SD",
+  "NE", "SD", "SD",
+  "PD", "any", "PD",
+  "any", "PD", "PD",
+  "NE", "NE", "NE",
+  "any", "missing", "NE",
+  "missing", "any", "NE"
+)
+
+knitr::kable(list_resp)
+```
+
+```{r}
+ovr_cbor <- covr_r11 %>%
+  derive_vars_transposed(
+    ovr_ca125,
+    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, ADT, ANL01FL, ANL02FL),
+    key_var = PARAMCD,
+    value_var = AVALC
+  ) %>%
+  select(-AVAL) %>%
+  rename(OVRR11 = AVALC) %>%
+  mutate(AVALC = case_when(
+    # CR
+    OVRR11 == "CR" & OVRCA125 == "CR" ~ "CR",
+    OVRR11 == "CR" & OVRCA125 %in% c("PR", "SD", "NE") ~ "PR",
+    OVRCA125 == "CR" & OVRR11 %in% c("PR", "SD", "NE") ~ "PR",
+    # PR
+    OVRR11 == "PR" & OVRCA125 %in% c("CR", "PR", "SD", "NE") ~ "PR",
+    OVRCA125 == "PR" & OVRR11 %in% c("CR", "PR", "SD", "NE") ~ "PR",
+    # SD, NE
+    OVRR11 %in% c("SD", "NE") & OVRCA125 %in% c("SD", "NE") & OVRR11 == OVRCA125 ~ OVRR11,
+    OVRR11 %in% c("SD", "NE") & OVRCA125 %in% c("SD", "NE") & OVRR11 != OVRCA125 ~ "SD",
+    # PD
+    OVRR11 == "PD" | OVRCA125 == "PD" ~ "PD",
+    # next assessment NA
+    is.na(OVRR11) | is.na(OVRCA125) ~ "NE",
+    TRUE ~ "MISSING"
+  ))
+
+ovr_cbor <- ovr_cbor %>%
+  # select(-OVRR11, -OVRCA125) %>%
+  mutate(
+    PARAMCD = "COVR11CA",
+    PARAM = "Combined Overall Response by Investigator (Confirmation of RECIST1.1 Reqired)",
+    PARAMN = 8,
+    PARCAT1 = "Combined",
+    PARCAT1N = 3,
+    PARCAT2 = "Investigator",
+    PARCAT2N = 1,
+    AVAL = aval_resp(AVALC)
+  )
+
+adrs <- bind_rows(adrs, ovr_cbor)
```

Comment:
This can be simplified by using `derive_param_computed()`:
```suggestion
adrs <- adrs %>%
  derive_param_computed(
    filter = CA125EFL == "Y" & ANL01FL == "Y" & ANL02FL == "Y",
    by_vars = exprs(!!!get_admiral_option("subject_keys"), AVISIT, ADT, ANL01FL, ANL02FL),
    parameters = c("OVRR11", "OVRCA125"),
    set_values_to = exprs(
      AVALC = case_when(
        # CR
        AVALC.OVRR11 == "CR" & AVALC.OVRCA125 == "CR" ~ "CR",
        AVALC.OVRR11 == "CR" & AVALC.OVRCA125 %in% c("PR", "SD", "NE") ~ "PR",
        AVALC.OVRCA125 == "CR" & AVALC.OVRR11 %in% c("PR", "SD", "NE") ~ "PR",
        # PR
        AVALC.OVRR11 == "PR" & AVALC.OVRCA125 %in% c("CR", "PR", "SD", "NE") ~ "PR",
        AVALC.OVRCA125 == "PR" & AVALC.OVRR11 %in% c("CR", "PR", "SD", "NE") ~ "PR",
        # SD
        AVALC.OVRR11 == "SD" | AVALC.OVRCA125 == "SD" ~ "SD",
        # PD
        AVALC.OVRR11 == "PD" | AVALC.OVRCA125 == "PD" ~ "PD",
        # NE
        !is.na(AVALC.OVRR11) | !is.na(AVALC.OVRCA125) ~ "NE",
        TRUE ~ "MISSING"
      ),
      PARAMCD = "COVR11CA",
      PARAM = "Combined Overall Response by Investigator (Confirmation of RECIST1.1 Reqired)",
      PARAMN = 8,
      PARCAT1 = "Combined",
      PARCAT1N = 3,
      PARCAT2 = "Investigator",
      PARCAT2N = 1,
      AVAL = aval_resp(AVALC)
    ),
    keep_nas = TRUE
  )
```
I have also updated/simplified the conditions for SD and NE.